tutorials/composite_bo_with_hogp.ipynb

```diff
@@ -0,0 +1,646 @@
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "models_used = (\n",
+    "    \"rnd\",\n",
+    "    \"ei\",\n",
+    "    \"ei_hogp_cf\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Problem Description"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We use a simple test problem describing the concentration of pollutants after a chemical spill from [Astudillo & Frazier, ICML, '19](https://proceedings.mlr.press/v97/astudillo19a.html) defined over a $3 \\times 4$ grid of values $s,t$ and we wish to optimize the parameters w.r.t. their true values, to estimate the true value of parameters, $x = [M, D, L, \\tau].$ The function is given by \n",
+    "$$ f(s,t | M, D, L, \\tau) := \\frac{M}{\\sqrt{4 \\pi D t}}  \\exp\\{-\\frac{s^2}{4Dt}\\} + \\frac{1_{t > \\tau} M}{\\sqrt{4 \\pi D(t - \\tau)}} \\exp\\{- \\frac{(s - L)^2}{4 D (t - \\tau)}\\}, $$\n",
+    "with the cheap to evaluate, differentiable function given by $g(y):= \\sum_{(s,t) \\in S \\times T} (c(s, t|x_{\\text{true}}) - y)^2.$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def env_cfun(s, t, M, D, L, tau):\n",
+    "    c1 = M / torch.sqrt(4 * math.pi * D * t)\n",
+    "    exp1 = torch.exp(-(s ** 2) / 4 / D / t)\n",
+    "    term1 = c1 * exp1\n",
+    "    c2 = M / torch.sqrt(4 * math.pi * D * (t - tau))\n",
+    "    exp2 = torch.exp(-((s - L) ** 2) / 4 / D / (t - tau))\n",
+    "    term2 = c2 * exp2\n",
+    "    term2[torch.isnan(term2)] = 0.0\n",
+    "    return term1 + term2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Helper Functions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "These are helper functions for us to maximize the acquisition function and to get random points."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def gen_rand_points(bounds, num_samples):\n",
+    "    points_nlzd = torch.rand(num_samples, bounds.shape[-1]).to(bounds)\n",
+    "    return bounds[0] + (bounds[1] - bounds[0]) * points_nlzd\n",
+    "    \n",
+    "\n",
+    "def optimize_ei(qEI, bounds, **options):\n",
+    "    with gpt_settings.fast_computations(covar_root_decomposition=False):\n",
+    "        cands_nlzd, _ = optimize_acqf(\n",
+    "                qEI, bounds, **options, \n",
+    "        )\n",
+    "    return cands_nlzd"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# ZEROISH = 1e-2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Below is a wrapped function to help us define bounds on the parameter space, we can also vary the size of the grid if we'd like to."
```

Comment:
should talk about how this constructs a `GenericMCObjective` and how it will be used